daemon/analytics.c

```diff
@@ -0,0 +1,833 @@
+    analytics_set_data (&analytics_data.netdata_alarms_critical,            "null");
+    analytics_set_data (&analytics_data.netdata_charts_count,               "null");
+    analytics_set_data (&analytics_data.netdata_metrics_count,              "null");
+    analytics_set_data (&analytics_data.netdata_config_is_parent,           "null");
+    analytics_set_data (&analytics_data.netdata_config_hosts_available,     "null");
+    analytics_set_data (&analytics_data.netdata_host_cloud_available,       "null");
+    analytics_set_data (&analytics_data.netdata_host_aclk_implementation,   "null");
+    analytics_set_data (&analytics_data.netdata_host_aclk_available,        "null");
+    analytics_set_data (&analytics_data.netdata_host_agent_claimed,         "null");
+
+    analytics_data.prometheus_hits = 0;
+    analytics_data.shell_hits = 0;
+    analytics_data.json_hits = 0;
+    analytics_data.dashboard_hits = 0;
+
+    char *default_port = appconfig_get(&netdata_config, CONFIG_SECTION_WEB, "default port", NULL);
+    int clean = 0;
+    if (!default_port) {
+        default_port = strdupz("19999");
+        clean = 1;
+    }
+
+    setenv("NETDATA_LISTEN_PORT"      , default_port, 1);
+    if(clean)
+        freez(default_port);
+
+    get_system_timezone();
+
+    // set the path we need
+    char path[1024 + 1], *p = getenv("PATH");
+    if(!p) p = "/bin:/usr/bin";
+    snprintfz(path, 1024, "%s:%s", p, "/sbin:/usr/sbin:/usr/local/bin:/usr/local/sbin");
+    setenv("PATH", config_get(CONFIG_SECTION_PLUGINS, "PATH environment variable", path), 1);
+
+    // python options
+    p = getenv("PYTHONPATH");
+    if(!p) p = "";
+    setenv("PYTHONPATH", config_get(CONFIG_SECTION_PLUGINS, "PYTHONPATH environment variable", p), 1);
+
+    // disable buffering for python plugins
+    setenv("PYTHONUNBUFFERED", "1", 1);
+
+    // switch to standard locale for plugins
+    setenv("LC_ALL", "C", 1);
+}
+
+void send_statistics( const char *action, const char *action_result, const char *action_data) {
+    static char *as_script;
+
+    if (netdata_anonymous_statistics_enabled == -1) {
+        char *optout_file = mallocz(sizeof(char) * (strlen(netdata_configured_user_config_dir) +strlen(".opt-out-from-anonymous-statistics") + 2));
+        sprintf(optout_file, "%s/%s", netdata_configured_user_config_dir, ".opt-out-from-anonymous-statistics");
+        if (likely(access(optout_file, R_OK) != 0)) {
+            as_script = mallocz(sizeof(char) * (strlen(netdata_configured_primary_plugins_dir) + strlen("anonymous-statistics.sh") + 2));
+            sprintf(as_script, "%s/%s", netdata_configured_primary_plugins_dir, "anonymous-statistics.sh");
+            if (unlikely(access(as_script, R_OK) != 0)) {
+               netdata_anonymous_statistics_enabled=0;
+               info("Anonymous statistics script %s not found.",as_script);
+               freez(as_script);
+            } else {
+               netdata_anonymous_statistics_enabled=1;
```

Comment:
Please, add space between equal, name and value.

Reply:
Ok, done!